src/Bicep.Core.IntegrationTests/TestFrameworkTests.cs

```diff
@@ -0,0 +1,97 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+using System;
+using System.Diagnostics.CodeAnalysis;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using Bicep.Core.Diagnostics;
+using Bicep.Core.Extensions;
+using Bicep.Core.Features;
+using Bicep.Core.Semantics;
+using Bicep.Core.Semantics.Namespaces;
+using Bicep.Core.TypeSystem;
+using Bicep.Core.UnitTests;
+using Bicep.Core.UnitTests.Assertions;
+using Bicep.Core.UnitTests.Utils;
+using Bicep.Core.Workspaces;
+using FluentAssertions;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace Bicep.Core.IntegrationTests
+{
+    [TestClass]
+    public class TestFrameworkTests
+    {
+        private ServiceBuilder ServicesWithTestFramework => new ServiceBuilder()
+            .WithFeatureOverrides(new(TestContext, TestFrameworkEnabled: true));
+
+        [NotNull]
+        public TestContext? TestContext { get; set; }
+
+        [TestMethod]
+        public void TestFramework_is_disabled_unless_feature_is_enabled()
+        {
+            var result = CompilationHelper.Compile(@"
+test test1 'test1.bicep' = {}
```

Comment:
I'm curious about why this syntax looks the way it does. Will the actual test assertions live in test1.bicep? If so, what is being assigned with the `= {}`? If not, why the reference to test1.bicep?

Is there a design doc or anything similar you could share?

Reply:
Hi, this is the document that Anthony created at the beginning of the internship.
https://gist.github.com/anthony-c-martin/74b3b7006d9e5fb303a8ce1aa735244b

Reply:
Nvm Jonny mentioned we will change the assertion declaration keyword to `assertion`, so we should be fine.